### PR TITLE
chore: update to node 14

### PR DESCRIPTION
**Pull request by chmelevskij**
Update to node 14. Also added `appdmg` to the latest resolutions to which uses latests `fs-xattr` which works on 14

### CODE REVIEW

**Comment by McGiverGim** [comment]
You must modify the readme.md file too, the development paragraph states to use Node 10.

**Comment by mikeller** [comment]
This should be done in unison with the same update in https://github.com/betaflight/blackbox-log-viewer, as they are normally released together.

**Comment by chmelevskij** [comment]
Done:
https://github.com/betaflight/blackbox-log-viewer/pull/495

**Comment by haslinghuis** [comment]
@chmelevskij testing now. Are there any additional steps for developers to implement this change locally?

```
[mark@localhost betaflight-configurator]$ yarn install
yarn install v1.22.5
[1/5] Validating package.json...
error betaflight-configurator@10.8.0: The engine "node" is incompatible with this module. Expected version "14.x". Got "10.23.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
[mark@localhost betaflight-configurator]$ yarn start
yarn run v1.22.5
error betaflight-configurator@10.8.0: The engine "node" is incompatible with this module. Expected version "14.x". Got "10.23.1"
error Commands cannot run with an incompatible environment.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by chmelevskij** [comment]
@haslinghuis don't think so

File: .travis.yml
Comment (mikeller) [inline .travis.yml:27]: This should stay in - Chrome is needed to make the tests work.
(Travis is not used by the project at the moment due to the lack of free machine time for open source projects.)

File: .travis.yml
Comment (chmelevskij) [inline .travis.yml:17]: @mikeller it's there. This was technically invalid yaml because there were 2 `addons` keys on the same level, so duplicate keys in dictionary pretty much.

**Comment by chmelevskij** [comment]
Looks like `yarn` doesn't like `resolutions` and `optionalDependencies` 😞  Unless people are happy to always use `--ignore-optional` which is not the most pleasant experience (if you are on windows or linux though)

File: .travis.yml
Comment (mikeller) [reply .travis.yml:17]: :+1:

**Review by mikeller — APPROVED** [review]

**Comment by mikeller** [comment]
For some reason the Windows and Linux builds are failing with this.

**Review by McGiverGim — DISMISSED** [review]
I continue thinking you need to change the `README.md` file in this PR.

image

**Comment by McGiverGim** [comment]
> For some reason the Windows and Linux builds are failing with this.

It seems the optional-dependencies are ok (it fails with a simple warning), but the resolutions not. It includes it for all.

**Comment by chmelevskij** [comment]
Yeah, will have to look into this. Might just fork the package which is causing headache and bring it up to node 14 deps

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> PASS
 - PR is not in draft -> PASS
 - Assigned to a milestone -> PASS
 - cooling off period lapsed (3/03/2021 06:49:57) -> PASS
 - commit count less or equal to three (1 commits) -> PASS
 - 'Don't merge' label NOT found -> PASS
 - 'Needs coordination with betaflight' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> FAIL
 - assigned to an approver (None) -> FAIL
 - approver count at least three (1 approvers) -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> FAIL
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
@chmelevskij - I can do the documentation if you like. Anything else to make this work?

**Comment by chmelevskij** [comment]
Current show stopper is https://github.com/aluxian/gulp-appdmg , I've tried forking it and upgrading the packages, but then we need to upgrade gulp version as well. So it's a bit of a cascading issue at the moment.

**Comment by haslinghuis** [comment]
```
[mark@localhost betaflight-configurator]$ gulp --version
CLI version: 2.3.0
Local version: 4.0.2
```

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
@chmelevskij Install Fedora 34 and found out `appdmg` is an `OSX` package. Did you try to update `gulp` as well?

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: README.md
Comment (chmelevskij) [inline README.md:69]: @McGiverGim what do you think? A bit worse ergonomics, but it won't be stale with version bumps.

Comment (haslinghuis) [reply README.md:69]: https://github.com/betaflight/betaflight-configurator/pull/2685

**Review by haslinghuis — APPROVED** [review]

File: README.md
Comment (haslinghuis) [reply README.md:69]: Just awaiting this change to get it updated.

**Comment by ctzsnooze** [comment]
The 9e395aba build seems to run normally on Mac OSX.

What code needs to be changed to get back to normal hide behaviour?  We currently configure cmd-H or 'hide' to result in Minimise, not hide.  I'm keen to test how we can get normal behaviour again.

**Comment by haslinghuis** [comment]
Tested on Windows 10.

**Comment by haslinghuis** [comment]
@chmelevskij LGTM! Can you please squash commits?

**Review by haslinghuis — APPROVED** [review]

File: .nvmrc
Comment (Asizon) [inline .nvmrc:1]: Seems that we need to use 14.5 or above to make it compatible with some module.

**Review by Asizon — CHANGES_REQUESTED** [review]

File: .nvmrc
Comment (chmelevskij) [reply .nvmrc:1]:
```suggestion
14.18.2
```

**Review by Asizon — APPROVED** [review]
Good work @chmelevskij

**Review by haslinghuis — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Comment by limonspb** [comment]
@McGiverGim could you please give a final look and merge? THanks!

**Review by ctzsnooze — APPROVED** [review]

**Pull request merged**
